File: misc/readfile/readfile.go
// +build ignore

package main

import "fmt"
import "io"
import "os"

func main() {
	if len(os.Args) < 2 {
		fmt.Printf("No file specified as first parameter.\n")
		os.Exit(1)
	}
	f, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Printf("%v\n", err)
		os.Exit(1)
	}
	fi, _ := f.Stat()
	buf := make([]byte, fi.Size())
	io.ReadFull(f, buf)
	f.Close()
	fmt.Printf("%v", string(buf))
	buf = nil
}

File: misc/readfile/readfile.c
// +build ignore

#include <stdio.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdlib.h>
#include <errno.h>

int main(int argc, char *argv[]) {
	int fd; long fs; char *buf;
	if (argc < 2) {
		printf("No file specified as first parameter.\n");
		return 1;
	}
	fd = open(argv[1], O_RDONLY);
	if (fd == -1) {
		perror(NULL);
		return errno;
	}
	fs = lseek(fd, 0, SEEK_END);
	lseek(fd, 0, SEEK_SET);
	buf = malloc(fs+1);
	read(fd, buf, fs);
	close(fd);
	write(STDOUT_FILENO, buf, fs);
	free(buf);
}
